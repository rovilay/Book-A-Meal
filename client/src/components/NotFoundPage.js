import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div className="main-container">
    <div className="setmenu-container">
      <div className="notFoundCard">
        <h1>404!</h1>
        You seem Lost!
        Click <Link to="/">here</Link> to go home
      </div>
    </div>
  </div>
);

export default NotFound;
