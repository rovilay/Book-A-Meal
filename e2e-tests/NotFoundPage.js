import 'babel-polyfill';

module.exports = {
  beforeEach: (browser) => {
    browser.maximizeWindow();
  },
  'It should redirect to `not found page` if route do not exist': (browser) => {
    browser
      .url('http://localhost:9000')
      .waitForElementVisible('body', 1000)
      .pause(3000)
      .url('http://localhost:9000/non-existing-page')
      .pause(2000)
      .waitForElementVisible('.not-found-page', 1000)
      .assert.containsText('#root > div > div:nth-child(1) > div > div > h1', '404!');
  },
  'It should redirect back home if `here` is clicked': (browser) => {
    browser
      .pause(2000)
      .click('#root > div > div:nth-child(1) > div > div > a')
      .pause(1000)
      .waitForElementVisible('body', 1000)
      .assert.containsText('#root > div > div:nth-child(1) > div > main > section.first-section > div > p', 'Meals that perfectly fits your lifestyle')
      .pause(2000)
      .end();
  }
};
