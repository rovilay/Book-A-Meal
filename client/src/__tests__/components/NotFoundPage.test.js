import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from '../../components/NotFoundPage';


describe('NotFoundPage component test', () => {
  it('should render NotFoundPage correctly', (done) => {
    const wrapper = shallow(<NotFoundPage />);

    expect(wrapper).toMatchSnapshot();

    done();
  });
});

