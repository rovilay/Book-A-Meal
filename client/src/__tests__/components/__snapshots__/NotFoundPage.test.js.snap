// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component test should render NotFoundPage correctly 1`] = `
<div
  className="not-found-page-container"
>
  <div
    className="not-found-page"
  >
    <h1>
      404!
    </h1>
    You seem Lost! Click 
    <Link
      replace={false}
      to="/"
    >
      here
    </Link>
     to go home.
  </div>
</div>
`;
